src/main/java/org/truffleruby/interop/ForeignReadStringCachedHelperNode.java

```diff
@@ -65,25 +64,9 @@ public Object readInstanceVariable(
 
     @Specialization(guards = {
             "!isRubyArray(receiver)", "!isRubyHash(receiver)", "!isIVar",
-            "methodDefined(frame, receiver, stringName, getDefinedNode())",
-            "!methodDefined(frame, receiver, INDEX_METHOD_NAME, getIndexDefinedNode())"
-    })
-    public Object callMethod(
-            VirtualFrame frame,
-            DynamicObject receiver,
-            Object name,
-            Object stringName,
-            boolean isIVar,
-            @Cached("create()") ForeignToRubyNode nameToRubyNode) {
-        return getCallNode().call(frame, receiver, METHOD_NAME, nameToRubyNode.executeConvert(name));
-    }
-
-    @Specialization(guards = {
-            "!isRubyArray(receiver)", "!isRubyHash(receiver)", "!isIVar",
-            "methodDefined(frame, receiver, stringName, getDefinedNode())",
             "methodDefined(frame, receiver, INDEX_METHOD_NAME, getIndexDefinedNode())"
     })
-    public Object callMethodPriorityOverIndex(
```

Comment:
This name was particularly confusing since it called `[]` but the method name said it called the method.